src/components/fields/switch/switch.tsx

```diff
@@ -0,0 +1,94 @@
+import { Form } from "@lifesg/react-design-system/form";
+import { Toggle } from "@lifesg/react-design-system/toggle";
+import { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import * as Yup from "yup";
+import { TestHelper } from "../../../utils";
+import { useValidationConfig } from "../../../utils/hooks";
+import { IGenericFieldProps } from "../../frontend-engine";
+import { FlexWrapper } from "./switch.styles";
+import { ISwitchSchema } from "./types";
+
+export const Switch = (props: IGenericFieldProps<ISwitchSchema>) => {
+	// =============================================================================
+	// CONST, STATE, REFS
+	// =============================================================================
+	const {
+		schema: { label, disabled, validation, customOptions, ...otherSchema },
+		id,
+		value,
+		error,
+		onChange,
+	} = props;
+
+	const [stateValue, setStateValue] = useState<boolean>(value || undefined);
+	const { setFieldValidationConfig } = useValidationConfig();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		setFieldValidationConfig(id, Yup.boolean(), validation);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [validation]);
+
+	useEffect(() => {
+		setStateValue(value);
+	}, [value]);
+
+	// =============================================================================
+	// EVENT HANDLERS
+	// =============================================================================
+	const handleClick = (value: boolean) => {
+		//console.log("click", stateValue, value);
+		onChange({ target: { value } });
+	};
+
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const isSwitchChecked = (value: boolean) => {
+		return stateValue === value;
+	};
+
+	const formatId = (type: string) => {
+		return `${id}-${type}`;
+	};
+
+	return (
+		<Form.CustomField id={id} label={label} errorMessage={error?.message}>
+			<FlexWrapper>
+				<Toggle
+					{...otherSchema}
+					key={formatId("yes")}
+					type="yes"
+					id={formatId("yes")}
+					data-testid={TestHelper.generateId(id, "switch-yes")}
+					disabled={disabled}
+					styleType={customOptions?.border === false ? "no-border" : "default"}
+					checked={isSwitchChecked(true)}
+					onChange={() => handleClick(true)}
+					indicator={true}
+					name="Yes"
```

Comment:
radio buttons in the same group should have the same `name` value. you can use `id` as the value for both toggles

Reply:
can i check other than aligning with the HTML standard, what functional behaviour this would have if we change the name? (Currently, making sure only one option is selected is managed by the react state. and this is following the implementation done in other radio buttons.)

Reply:
- enables arrow key navigation between the two options
- on Voiceover, it reads out "Yes, radio button, **1 of 2**" instead of "1 of 1", which lets the user know that there are multiple options in the field

Reply:
Done